src/main/java/reposense/model/AuthorConfiguration.java

```diff
@@ -53,16 +54,32 @@ public void update(StandaloneConfig standaloneConfig, List<String> ignoreGlobLis
             newAuthorDisplayNameMap.put(author, author.getDisplayName());
             List<String> aliases = new ArrayList<>(author.getAuthorAliases());
             List<String> emails = new ArrayList<>(author.getEmails());
+            allAliases.addAll(aliases);
             aliases.add(author.getGitId());
+
             aliases.forEach(alias -> newAuthorEmailsAndAliasesMap.put(alias, author));
```

Comment:
Just to clarify, is there a reason we're not doing the check for duplicates as we add the alias into the TreeMap?
The `TreeMap.put` method returns null if the key is a new key

Reply:
I agree with your suggestion that we can do it as we add the alias into the TreeMap. But the thing is it becomes a bit irrelevant. Cause even if the alias is duplicate, it will still have to be added, as the implementation decided is that the last person to claim the alias get it. 

Also, if I do it inside the function, then I will still have to check duplicates for CSV files. To prevent this duplication, I am doing it outside. This way I can use the same function for both.

Reply:
> Cause even if the alias is duplicate, it will still have to be added, as the implementation decided is that the last person to claim the alias get it.

I don't quite see the issue. If we want, we can change the implementation such that we check whether each alias exists before adding it, and prevent existing aliases from being overwritten.

> Also, if I do it inside the function, then I will still have to check duplicates for CSV files. To prevent this duplication, I am doing it outside.

What if we create a new static method that accepts a TreeMap, a key and a value, and then:
- adds the key - value pair to the TreeMap
- checks if the key already existed by inspecting the return value of `TreeMap.put`
- uses the logger to issue a warning if the key already existed?

We can use this method for parsing both csv and standalone config?

Reply:
I am okay with changing the implementation. But I believe the senior devs and Prof in their original implementation wanted it to be overwritten. Can ask for their suggestion on this. I am not hard bent on any fixed way, but think last person should to claim alias should get it. 

@eugenepeh your suggestion for this?

Reply:
Sorry, let me clarify.
I'm not suggesting that we *need* to change the current implementation *now*.
I'm suggesting that if we perform the check for duplicates as we add each alias into the TreeMap, then it'll be easier for us to change the handling of duplicates *if we want, in future*.